packages/nodes-base/nodes/HomeAssistant/HomeAssistant.node.ts

```diff
@@ -0,0 +1,395 @@
+		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
+		description: 'Consume Home Assistant Io API',
+		defaults: {
+			name: 'Home Assistant',
+			color: '#3578e5',
+		},
+		inputs: [ 'main' ],
+		outputs: [ 'main' ],
+		credentials: [
+			{
+				name: 'homeAssistantApi',
+				required: true,
+			},
+		],
+		properties: [
+			{
+				displayName: 'Resource',
+				name: 'resource',
+				type: 'options',
+				options: [
+					{
+						name: 'Camera Proxy',
+						value: 'cameraProxy',
+					},
+					{
+						name: 'Config',
+						value: 'config',
+					},
+					{
+						name: 'Event',
+						value: 'event',
+					},
+					{
+						name: 'History',
+						value: 'history',
+					},
+					{
+						name: 'Log',
+						value: 'log',
+					},
+					{
+						name: 'Service',
+						value: 'service',
+					},
+					{
+						name: 'State',
+						value: 'state',
+					},
+					{
+						name: 'Template',
+						value: 'template',
+					},
+				],
+				default: 'config',
+				description: 'Resource to consume.',
+			},
+			// Camera proxy
+			...cameraProxyOperations,
+			...cameraProxyFields,
+			// Configuration
+			...configOperations,
+			...configFields,
+			// Event
+			...eventOperations,
+			...eventFields,
+			// History
+			...historyOperations,
+			...historyFields,
+			// Log
+			...logOperations,
+			...logFields,
+			// Service
+			...serviceOperations,
+			...serviceFields,
+			// State
+			...stateOperations,
+			...stateFields,
+			// Template
+			...templateOperations,
+			...templateFields,
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+		const length = items.length as unknown as number;
+		const resource = this.getNodeParameter('resource', 0) as string;
+		const operation = this.getNodeParameter('operation', 0) as string;
+		const qs: IDataObject = {};
+		let responseData;
+		for (let i = 0; i < length; i++) {
+			try {
+				if (resource === 'config') {
+					if (operation === 'get') {
+						responseData = await homeAssistantApiRequest.call(this, 'GET', '/config');
+					} else if (operation === 'check') {
+						responseData = await homeAssistantApiRequest.call(this, 'POST', '/config/core/check_config');
+					}
+				} else if (resource === 'service') {
+					if (operation === 'getAll') {
+						const returnAll = this.getNodeParameter('returnAll', i) as boolean;
+						responseData = await homeAssistantApiRequest.call(this, 'GET', '/services');
+						if (!returnAll) {
+							const limit = this.getNodeParameter('limit', i) as number;
+							responseData = (responseData as IDataObject[]).slice(0, limit);
+						}
+					} else if (operation === 'call') {
+						const domain = this.getNodeParameter('domain', i) as string;
+						const service = this.getNodeParameter('service', i) as string;
+						const additionalFields = this.getNodeParameter('additionalFields', i) as IDataObject;
```

Comment:
Instead of `IDataObject`, you can type this exactly to clear out all the castings below.